sdk/src/azure/core/az_mqtt5_connection_hfsm.c

```diff
@@ -218,6 +240,13 @@ static az_result started(az_event_policy* me, az_event event)
           az_event_policy_send_inbound_event((az_event_policy*)this_policy, event));
       break;
 
+    case AZ_HFSM_EVENT_ERROR:
+    case AZ_HFSM_EVENT_TIMEOUT:
+      _az_RETURN_IF_FAILED(_az_hfsm_transition_peer((_az_hfsm*)me, started, faulted));
+      _az_RETURN_IF_FAILED(
+          az_event_policy_send_inbound_event((az_event_policy*)this_policy, event));
```

Comment:
On both timeout and error, the event sent inbound should be ERROR, not timeout. This is done for two purposes:
1. Inform all sub-state machines that an unhandled error occurred in connection. Likely, all state machines now need to switch to faulted since they depend on connectivity.
2. The error_data struct contains fields indicating which policy and original event caused the error - good for debugging.